File: 3042-Count-Prefix-and-Suffix-Pairs-I.java
class Solution {
    public int countPrefixSuffixPairs(String[] words) {
        int pairs = 0;
        int N = words.length;
        for(int i =0; i < N-1 ;i ++){
            for(int j = i+1 ; j < N ; j++){
                if((words[i].length() <= words[j].length()) && isPrefixandSuffix(words[i],words[j])){
                    pairs ++;
                }
            }
        }
        return pairs;
    }
    public boolean isPrefixandSuffix(String word1, String word2){
        return word2.startsWith(word1) && word2.endsWith(word1);
    }
}

File: 3042-Count-Prefix-and-Suffix-Pairs-I.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        count = 0
        for i in range(len(words)-1):
            for j in range(i+1, len(words)):
                if len(words[j]) < len(words[i]):
                    continue 
                if words[j].startswith(words[i]) and words[j].endswith(words[i]):
                    count += 1
        return count
